### PR TITLE
Fixed #1203: Migrate tests from the Scala to the plain Java

### LINKED ISSUE #1203: Migrate tests from the Scala to the plain Java
Rework all test to the Java with the TestNG framework

- [x] add required dependencies to the pom.xml files
- [ ] swagger-core #1206 
- [x] swagger-jaxrs #1207 
- [x] swagger-jersey2-jaxrs #1208 
- [ ] swagger-hibernate-validations #1209

### PR DESCRIPTION
**Pull request by iushankin**
- Added configuration for the TestNG
- Created sample test

### CODE REVIEW

File: modules/swagger-core/src/test/java/io/swagger/BeanValidatorTestNG.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import io.swagger.models.properties.IntegerProperty;
+import io.swagger.models.properties.Property;
+import io.swagger.models.properties.StringProperty;
+import models.BeanValidationsModel;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+
+public class BeanValidatorTestNG extends Assert {
```
Comment (olensmar) [inline modules/swagger-core/src/test/java/io/swagger/BeanValidatorTestNG.java]: I think just "BeanValidatorTest" is fine... :-)

File: pom.xml [hunk 2]
```diff
@@ -47,7 +47,6 @@
     <build>
         <sourceDirectory>src/main/scala</sourceDirectory>
         <scriptSourceDirectory>src/main/scripts</scriptSourceDirectory>
-        <testSourceDirectory>src/test/scala</testSourceDirectory>
```
Comment (olensmar) [inline pom.xml]: do we really want to remove the scala tests before all of them are migrated?

Comment (iushankin) [inline pom.xml]: The ScalaTest compiles tests from the src/test/scala path by default.
We compile tests from both paths: src/test/scala, src/test/java.
Please look at build.log file - https://travis-ci.org/swagger-api/swagger-core/builds/68503343#L3622-L3623

File: modules/swagger-core/src/test/java/io/swagger/BeanValidatorTestNG.java (hunk 1 above)
Comment (iushankin) [inline modules/swagger-core/src/test/java/io/swagger/BeanValidatorTestNG.java]: Yes, but we have some java tests in scala (i.e. SimpleGenerationTest, ModelPropertyOverrideTest, etc.). And these tests will be run two times (first run by the ScalaTest and second run by the TestNG).
It's best workaround when we have two types of tests.

Comment (olensmar) [inline modules/swagger-core/src/test/java/io/swagger/BeanValidatorTestNG.java]: ok - we can rename them once the Scala tests have removed... thanks

**Comment by olensmar** [comment]
thank you!

File: modules/swagger-core/src/test/java/io/swagger/BeanValidatorTestNG.java (hunk 1 above)
Comment (tomtit) [inline modules/swagger-core/src/test/java/io/swagger/BeanValidatorTestNG.java]: I don't like the idea with renaming and TestNG ending. There is nothing wrong if some tests will run twice (of course until they will not fail). This just means that these tests are first candidates for migration.
But my main concern here that we introduce implicit naming convention for Java tests for other community members.

**Pull request merged**